### DISCUSSION TITLE
Should I create a dedicated table for "statuses"?

### DISCUSSION THREAD

**Discussion by Vadorequest** [supabase/supabase#620] [Questions] [answered]
Using Airtable, I have a choice when I want to create a column where the value can be selected within a list:
1. It can be a "hardcoded" list, which is stored in the column itself. This is simpler but less flexible, as each record basically stores a value, not a reference
1. It can be a relationship to another table. This requires creating a dedicated table to store all the values. It's more flexible because it can be extended (add more columns).

Using Postgre, I feel like the easy solution isn't obvious, maybe not even possible? 

For instance, I want each "Project" to have a "Status", I went for the 2nd solution.
![image](https://user-images.githubusercontent.com/3807458/106029326-41986280-60cd-11eb-97f9-eb4c618cdb92.png)

![image](https://user-images.githubusercontent.com/3807458/106029354-49580700-60cd-11eb-962d-523a1624d22f.png)

It's not really user-friendly either, because the UI could display a list of elements but it doesn't, I have to manually type a proper value. I feel it's overkill for such a simple use case and wonder if I can simplify it by avoiding creating a dedicated table for the "Project statuses".

**Comment by steve-chavez** [comment, marked as answer]
1 is possible with PostgreSQL, by using an [ENUM](https://www.postgresql.org/docs/current/datatype-enum.html) type.

I think right now it isn't possible to create enums within the Table Editor, but it can be added(cc @joshenlim).

As a workaround, you can create the enum with SQL:

```sql
CREATE TYPE "ProjectStatus" AS ENUM ('active', 'closed', 'etc');

ALTER TABLE "Project" ADD COLUMN status "ProjectStatus";
```

Then you could use the enum values through the Table Editor.

**Reply by Vadorequest** [reply]
It did work, thanks for the tip.

Although, I had to refresh the Editor for the column to appear, and it doesn't propose a list of valid values either, but those are different issues.
